pandas/tests/io/parser/test_c_parser_only.py

```diff
@@ -176,6 +176,9 @@ def test_precise_conversion(c_parser_only):
     normal_errors = []
     precise_errors = []
 
+    def error(val, actual_val):
+        return abs(Decimal(f"{val:.100}") - actual_val)
```

Comment:
can we add type hints (both here and to the `_isin` function you've added) whilst we're here? any new development should ideally include them

other than that, looks good to me

Reply:
Changed, ran `pre-commit`, pushed change.